cloudbridge/cloud/providers/azure/azure_client.py

```diff
@@ -585,13 +587,52 @@ def delete_subnet(self, subnet_id):
                                              subnet_id)
         network_name = url_params.get(NETWORK_NAME)
         subnet_name = url_params.get(SUBNET_NAME)
-        result_delete = self.network_management_client \
-            .subnets.delete(
-                self.resource_group,
-                network_name,
-                subnet_name
-            )
-        result_delete.wait()
+
+        try:
+            result_delete = self.network_management_client \
+                .subnets.delete(
+                    self.resource_group,
+                    network_name,
+                    subnet_name
+                )
+            result_delete.wait()
+        except CloudError as cloud_error:
+            log.exception(cloud_error.message)
+            print(cloud_error.code)
+            if cloud_error.code != "InUseSubnetCannotBeDeleted":
```

Comment:
Why does this error get triggered? In principle, our tests should wait for the previous object to be deleted before attempting to delete the subnet (using the resource.wait_for method), so the subnet should not be in use any longer?